frontend/app/ui/UploadForm/UploadForm.tsx

```diff
@@ -4,11 +4,13 @@ import Dropdown from "../Dropdown";
 import { FileUpload } from "../FileUpload/FileUpload";
 import { MultiSelectDropdown } from "../MultiSelectDropdown";
 import { useRouter } from "next/navigation";
+import { updateResults } from "@/lib/features/ResultsSlice";
+import { useAppDispatch } from "@/lib/hooks";
 import "./UploadForm.scss";
 
 export default function UploadForm() {
   const router = useRouter();
-
+  const dispatch = useAppDispatch();
   const formdata = new FormData();
   const uploadResume = async (formdata: FormData) => {
```

Comment:
Could you please make this method a server action? (by that we can hide our API endpoints & secrets)